code/obj/item/organs/stomach.dm

```diff
@@ -67,6 +111,53 @@
 				output += "[S] = [L[S]]\n"
 			boutput(user, "<br><span style='color:purple'><b>[src]</b> contains:\n [output]</span>")
 
+	relaymove(mob/user, direction, delay, running)
+		if(!prob(60) || !src.donor || !(user in src.stomach_contents))
+			return
+		src.donor.audible_message("<span class='alert'>You hear something rumbling inside [src.donor]'s stomach...</span>")
+		var/obj/item/I = user.equipped()
+		if(I?.force)
+			var/d = rand(round(I.force / 4), I.force)
+			src.donor.TakeDamage("chest", d, 0)
+			src.donor.visible_message("<span class='alert'><B>[user] attacks [src.donor]'s stomach wall with \the [I.name]!</span>")
+			playsound(user.loc, 'sound/impact_sounds/Slimy_Hit_3.ogg', 50, 1)
+
+			if(prob(src.donor.get_brute_damage() - 50))
+				logTheThing(LOG_COMBAT, user, "gibs [constructTarget(src.donor,"combat")] breaking out of their stomach at [log_loc(src.donor)].")
+				src.donor.gib()
+
+	proc/handle_digestion(mult = 1)
+		if (!length(src.stomach_contents))
+			return
+		src.digest_food(mult)
+		src.digest_mobs(mult)
+
+#define DIGESTION_PER_LIFE_TICK 3 //Total amount of reagents we can digest each Life tick
```

Comment:
seems like this should probably just be a var on the stomach, then you can have cyberstomachs process faster or whatever